test/test_utils.py

```diff
@@ -49,29 +51,15 @@ def test_pt_actx_key_stringification_uniqueness():
 
 def test_dataclass_array_container() -> None:
     from dataclasses import dataclass, field
-    from typing import Optional, Tuple  # noqa: UP035
 
     from arraycontext import Array, dataclass_array_container
 
-    # {{{ string fields
-
-    @dataclass
-    class ArrayContainerWithStringTypes:
-        x: np.ndarray
-        y: "np.ndarray"
-
-    with pytest.raises(TypeError, match="String annotation on field 'y'"):
-        # NOTE: cannot have string annotations in container
-        dataclass_array_container(ArrayContainerWithStringTypes)
-
-    # }}}
-
     # {{{ optional fields
 
     @dataclass
     class ArrayContainerWithOptional:
         x: np.ndarray
-        y: Optional[np.ndarray]
+        y: np.ndarray | None
```

Comment:
This was intentionally left as `Optional[...]` (same for the `Tuple` and `Union` below). E.g. `Tuple[...]` and `tuple[...]` have different base classes, so `dataclass_container` needs to check for both.

Reply:
Fixed!